File: app/src/main/java/tw/ctl/interest/MainActivity.java
package tw.ctl.interest;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import tw.ctl.interest.calculation.CalculationFragment;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.toolbar)    Toolbar toolbar;
    @BindView(R.id.view_pager) ViewPager viewPager;
    @BindView(R.id.tab_layout) TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);

        viewPager.setAdapter(new ViewPagerAdapter(getSupportFragmentManager()));

        tabLayout.setupWithViewPager(viewPager);

        /** Set each tab view. */
        for (int i = 0; i < tabLayout.getTabCount(); i++) {
            TabLayout.Tab tab = tabLayout.getTabAt(i);
            if (tab == null) continue;
            tab.setCustomView(getTabView(i));
        }
    }

    private View getTabView(int position) {
        View view = LayoutInflater.from(this).inflate(R.layout.tab, tabLayout, false);
        TabViewHolder viewHolder = new TabViewHolder(view);
        viewHolder.tabTitle.setText(position == 0 ? "計算" : (position == 1 ? "歷史" : "資訊"));
        Drawable drawable = ContextCompat
                .getDrawable(this, position == 0 ? R.drawable.tab_calculator :
                        (position == 1 ? R.drawable.tab_history : R.drawable.tab_info));
        drawable.setBounds(0, 0, drawable.getMinimumWidth(), drawable.getMinimumHeight());
        viewHolder.tabTitle.setCompoundDrawables(null, drawable, null, null);
        return view;
    }

    class ViewPagerAdapter extends FragmentPagerAdapter {

        ViewPagerAdapter(FragmentManager fragmentManager) {
            super(fragmentManager);
        }

        @Override
        public Fragment getItem(int position) {
            switch (position) {
                case 1:
                    return new HistoryFragment();
                case 2:
                    return new InfoFragment();
                default:
                    return new CalculationFragment();
            }
        }

        @Override
        public int getCount() {
            return 3;
        }

    }

    class TabViewHolder {

        @BindView(R.id.tab_title) TextView tabTitle;

        TabViewHolder(View view) {
            ButterKnife.bind(this, view);
        }

    }

}

File: app/src/main/java/tw/ctl/interest/MainActivity.kt
package tw.ctl.interest

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import kotlinx.android.synthetic.main.activity_main.*
import tw.ctl.interest.calculation.CalculationFragment
import tw.ctl.interest.history.HistoryFragment
import tw.ctl.interest.info.InfoFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(toolbar)

        bottomNavigationView.setOnNavigationItemSelectedListener { item: MenuItem ->
            selectFragment(item.itemId)
            true
        }

        selectFragment(0)
    }

    private fun selectFragment(itemId: Int) {
        when (itemId) {
            R.id.calculation -> {
                val fragment = supportFragmentManager.findFragmentByTag(CalculationFragment::class.java.simpleName)
                if (fragment == null) {
                    switchFragment(CalculationFragment())
                } else {
                    switchFragment(fragment)
                }
            }
            R.id.history -> {
                val fragment = supportFragmentManager.findFragmentByTag(HistoryFragment::class.java.simpleName)
                if (fragment == null) {
                    switchFragment(HistoryFragment())
                } else {
                    switchFragment(fragment)
                }
            }
            R.id.info -> {
                val fragment = supportFragmentManager.findFragmentByTag(InfoFragment::class.java.simpleName)
                if (fragment == null) {
                    switchFragment(InfoFragment())
                } else {
                    switchFragment(fragment)
                }
            }
            else -> {
                switchFragment(CalculationFragment())
            }
        }
    }

    private fun switchFragment(fragment: Fragment) {
        val fragments = supportFragmentManager.fragments

        val transaction = supportFragmentManager.beginTransaction()

        for (f in fragments) {
            transaction.hide(f)
        }

        if (fragment.isAdded) {
            transaction.show(fragment)
        } else {
            transaction.add(R.id.container, fragment, fragment.javaClass.simpleName)
        }

        transaction.commit()
    }

}
